### PR TITLE
Refactor `_create_custom_app_strings()`

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

I am going to need to add to app_strings.txt, and I noticed a `HELPME`, so I'm helping.

This PR refactors `_create_custom_app_strings()`.

Commits are broken down logically :blowfish: but I've mostly just pulled one 200-line function into a bunch of smaller functions, and added some line breaks, so maybe :office: :shrug: 

## Feature Flag

N/A

### Safety story

No change to functionality.

### Automated test coverage

Covered by corehq.apps.app_manager.tests.test_apps, among others.

### QA Plan

No QA planned.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/app_manager/app_strings.py [hunk 1]
```diff
@@ -214,6 +185,53 @@ def _create_custom_app_strings(app, lang, for_default=False, build_profile_id=No
         )
 
 
+def _create_search_app_strings(
```
Comment (sravfeyn) [inline corehq/apps/app_manager/app_strings.py:309]: Nit: I would perhaps use `_create_case_search_app_strings` to be explicit.

File: corehq/apps/app_manager/app_strings.py [hunk 2]
```diff
@@ -41,17 +41,6 @@ def _get_custom_icon_app_locale_and_value(custom_icon_form, custom_icon_text, fo
 
 
 def _create_custom_app_strings(app, lang, for_default=False, build_profile_id=None):
-    # HELPME
```
Comment (sravfeyn) [inline corehq/apps/app_manager/app_strings.py:44]: 💯

**Review by sravfeyn — APPROVED** [review]
Thanks, nice cleanup!

**Comment by Charl1996** [comment]
I agree with @sravfeyn , this is a very nice cleanup. 

One question, though: how do we know it still works? Is/was there a test for the original that you simply ran to verify the refactored version still works?

**Comment by kaapstorm** [comment]
@Charl1996 good question! I tested by commenting out some of the functions, and checking that tests failed. (corehq.apps.app_manager.tests.test_apps and corehq.apps.app_manager.tests.test_app_strings fail.) I checked my refactors of `if` statements, but I did not check _all_ functions -- I'm pretty confident about the functions I didn't check, because they are just cut-and-pasted, with some line breaks added.

**Review by Charl1996 — APPROVED** [review]

**Pull request merged**